scripts/generate_certs.py

```
@@ -0,0 +1,156 @@
+        filename: where value of variable stored
+        """
+        output = "const char " + name + "[] =\n"
+        with open(filename) as f:
+            for line in f.read().splitlines():
+                output += "    \"" + line + "\\r\\n\"\n"
+            output = output[:-1] + ";\n"
+        return output
+
+    @staticmethod
+    def type_macro_bin(name, filename):
+        """ Generates code snipper to be added
+        name: name of macro
+        filename: where value of macro stored
+        """
+        output = "#define " + name + " {\n"
+        data = subprocess.check_output(["xxd", "-i", filename]).decode("utf-8")
+        output += '  ' + '  '.join(data.splitlines(keepends=True)[1:-2])
+        output = "".join([(line + (77-len(line))*" " + "\\\n") for line in output.splitlines()])
+        output += "}\n"
+        return output
+
+    @staticmethod
+    def type_macro_string(name, filename):
+        """ Generates code snipper to be added
+        name: name of macro
+        filename: where value of macro stored
+        """
+        output = "#define " + name + "\n"
+        with open(filename) as f:
+            for line in f.read().splitlines()[:-1]:
+                output += "    \"" + line + "\\r\\n\"\n"
+            output = output[:-1] + "\n"
+        output = "".join([(line + (75-len(line))*" " + "\\\n") for line in output.splitlines()])
+        with open(filename) as f:
+            output += "    \"" + f.read().splitlines()[-1] + "\\r\\n\"\n"
+        return output
+
+    # Extract key or certificate from file.
+    def extract_key(self, enc, typ, name, filename):
+        """ Depending on encoding and type it calls appriopriate
+        function
+
+        enc: encoding
+        typ: type
+        name: name of variable/macro
+        filename: where data is
+        """
+        if enc not in ("string", "binary"):
+            print("choose either string encoding or binary encoding")
+            sys.exit(1)
```

Would potentially be better to raise a custom exception here, caught by a try/except block in Update(), rather than just sys.exit(). Certainly we should be reporting what the type passed in was.

---

added new custom exceptions. Catches it in main and sys exit from run_main() function